packages/web/src/press/PressPage.tsx

```diff
@@ -20,29 +20,43 @@ export interface PressArticleFields {
 
 interface Props {
   press: PressArticleFields[]
+  languages: string
 }
 
 class PressPage extends React.PureComponent<I18nProps & Props> {
   static async getInitialProps(context) {
     let press = []
+    let languages
     try {
       // when run on server import fetching code and run. on client send req to api
       if (context.req) {
         const getpress = await import('src/../server/fetchPress')
         press = await getpress.default()
+        languages = context.req.headers['accept-language']
       } else {
-        press = await fetch(`/api/press`).then((result) => result.json())
+        const res = await fetch(`/api/press`)
+        press = await res.json()
+        languages = res.headers.get('accept-language')
       }
-      return { press }
+      return { press, languages }
     } catch {
-      return { press }
+      return { press, languages }
     }
   }
   render() {
-    const { t, press, i18n } = this.props
+    const { t, press, languages, i18n } = this.props
+    const langList = new Set(
+      languages
```

Comment:
nit: consider moving the string parsing into the getInitialProps so it only needs to run once rather than on each render. 

keep the creation of the Set here though.